shub/apps/library/views/images.py

```diff
@@ -551,6 +554,60 @@ def get(self, request):
         collections = generate_collections_list(token.user)
         return Response(data=collections, status=200)
 
+    def post(self, request):
+        """Create a new collection.
+
+        POST /v1/collections
+
+        Body parameters:
+        * entity: entity id (= user id)
+        * name: new collection name
+        * private: optional boolean (defaults to the configured default for
+          new collections
+
+        Return the newly created collection.
+        """
+
+        print("POST CollectionsView")
+        if not validate_token(request):
+            print("Token not valid")
+            return Response(status=403)
+
+        # body should have {'entity': entity_id, 'name': new_collection_name, 'private': bool}
+        # 'private' is optional and defaults to Collection.private default
+        # value set with get_privacy_default()
+        # {"entity": "42", "name": "my_collection", "private": true}
```

Comment:
Although sylabs might use an entity ID, the user making the request isn't easily going to know their. So let's change this (and the subsequent query) to look for the user's `user.username`.

Reply:
But this commenting is great - by all means please leave the commenting for the developer to see!

Reply:
And if it's the case that Singularity uses another endpoint to get the ID first based on the user provided usrename and then issues this request to create the collection, then it would be ok to leave as is. I think we'd want to note in the docs (with the curl example perhaps?) How to look up the user id.

Reply:
Not sure what you mean. My aim is to replicate the Sylabs endpoint actually. It wouldn't make sense to change its signature.

Reply:
In the sregistry data model the Sylabs entity id corresponds to the user id.

Reply:
So what endpoint is the Sylabs client using to retrieve (or lookup) a user to get the id in the first place? Is this endpoint already supported in sregistry?

Reply:
I don't know which endpoint the Sylabs client uses, because I use CLI (bash + curl) with *this* endpoint  the Sylabs library, and I need the sregistry's endpoint to be the same.

Reply:
This is why you should test with the Singularity client. If an endpoint is needed to retrieve the user first, we'll add it here.

Reply:
Maybe you could explain what you mean by "retrieve the user"? Would it be "retrieve the entity id from its name"? In that case this can be done with the `/v1/entities/{user}` endpoint. It returns an entity object which features an `id` attribute.

BTW I am just a user of our Sylabs instance. I don't have any admin access and won't be granted one. Then I can't trace which endpoints are involved when the Sylabs client interact with the library.